### DISCUSSION TITLE
Device heating and app freezing issue on iOS devices

### DISCUSSION THREAD

**Discussion by codeclouds-team** [socketio/socket.io#4372] [Q&A]
I am developing an app that is built on using react-native 0.63 and facing a weird issue that when the app is running and socket connection is active it makes the device heat more than normal, and also if the app is opened from the background state the app scrolling works but other actions like navigation or swipe those become non-response and takes too much time to respond for a single action,
if we kill and restart the app then it works again but the same thing repeats again after keeping the app inactive for some time.
There is no error report too.

This is the socket manager class we have in our app

```
import SocketIOClient from 'socket.io-client';
class SocketManager {
    constructor() {
        this.socket = null;
        this.socket = {
            on : ()=>{},
            emit : ()=>{}
        }
         this.socket = SocketIOClient('https://mysocketserver.com',{
             reconnection:true,
             autoConnect : true
         });
         this.socket.on('connect', (res) => {
            //Updating socket id in the database for the user
         });
         this.socket.on('message',(data)=>{
             //Notifying
         })
    }
   
    _socketObj(){
        return this.socket;
    }
}
export default new SocketManager();
```

And I am calling this on index.js file of react-native (that is in the root of the project, and the first file which initializes on the app wake up) because I have calling feature in the app so to ensure that I get an active socket connection as soon as the app wakes up through the VOIP push notifications or firebase push notification
this is an example how I am calling it in index.js
```
import SocketManager from 'path to the class file/socket';
let socket = SocketManager._socketObj();
```

and surprisingly on Android nothing similar is happening, all is working fine on android.
Please suggest a solution.

**Comment by darrachequesne** [comment]
Hi! Are you sure this is related to the library? Creating a WebSocket connection should not consume that much memory/CPU.

**Reply by cisplRejjak** [reply]
The same heating problem here!

**Reply by codeclouds-team** [reply]
I have noticed that if we comment the socket connection code, then the heating issue is not happening at least, but the APP still does freeze. it has WebRTC based calling function too including the Callkeep library and VOIP push notification library to make the calling function work properly on iOS.
